### DISCUSSION TITLE
Feature Request: Support for expo-av

### DISCUSSION THREAD

**Discussion by petracles** [muxinc/elements#593] [Ideas]
Hey there - I was sent here by Mux Support when enquiring about Mux SDK support for [Expo's video player, expo-av](https://docs.expo.dev/versions/latest/sdk/video/). I was instructed to open a feature request, and I'm hoping to get some information on whether there'll be any support for expo-av in the future.

It's maybe worth noting that I also looked into replacing the expo-av player with react-native-video, but I found that in order to replace it, I would have to eject my project from Expo.

I did get some playback working when wrapping the expo-av player in the react-native-video Mux SDK. However, the main issue I am seeing is that the video player remounts when any state vars are updated in `onProgress` (or `onPlaybackStatusUpdate` with expo-av). This results in playback stuttering/looping as the player remounts on each poll.

I hope this info helps, and if there's anything I can do to help support expo-av with Mux, I'm more than happy to help. This issue really puts my work on halt, so any info on this is very appreciated. Thanks!

**Comment by dylanjha** [comment]
Hi @petracles thanks for the issue. `<mux-player>` and `<MuxPlayer />` React are both for web right now. But we would the equivalents on the native platforms iOS, Android, React Native, Flutter, etc.

> I did get some playback working when wrapping the expo-av player in the react-native-video Mux SDK

That is the [Mux Data SDK for react-native-video](https://github.com/muxinc/mux-stats-sdk-react-native-video)? Sounds like a bug in that SDK that we'll need to patch in that SDK to play nice with expo-av

**Reply by liquidvisual** [reply]
Did this ever get patched?

**Comment by petracles** [comment]
Hey @dylanjha thanks for the quick response.

Yeah correct, I'm referring to the Mux Data SDK for react-native-video. If there is an easy fix to make the expo-av player compatible, that would be _huge_ and would unblock my Mux Data integration for my project.

Let me know if there's anything I can do to help with this! Happy to share any source or test any fixes/suggestions you have in mind.

Thanks again!

**Reply by liquidvisual** [reply]
How are you going in 2024? We're looking at a possible expo-av integration after reading [this article](https://www.mux.com/blog/build-an-instagram-style-social-media-app-using-react-native-firebase-and-mux) but the issue you raised sounds like a pretty substantial blocker for us also.
